src/components/Dialog-Window/Notes/DialogNote.tsx

```diff
@@ -28,11 +28,20 @@ export const DialogNote: React.FC<NoteProps> = ({
   const [noteDone, setMarkedAsDone] = React.useState(
     userData[contentId]?.dialogs[id]?.noteDone ?? false,
   );
-  const [wordCount, setWordCount] = React.useState(0);
-  const [maxWordCount, setMaxWordCount] = React.useState<number | undefined>();
+  const [characterCount, setCharacterCount] = React.useState(0);
+  const maxLengthExceeded = maxLength ? characterCount > maxLength : false;
+  const characterCountText =
+    t('noteCharacterCountDescriptiveText')
+      .replace('@count', characterCount.toString())
+      .replace('@max', maxLength?.toString() ?? '');
```

Comment:
Perhaps we should use two translations here - one with and one without `@max`? The way it's laid out now, `@max` can't really be used, in case it's not set, as the translation would be something like `You have written 123 words. The max amount of words is .`

Reply:
Ah, I see - the text only shows when `maxLength` is set:) What do you think about doing something like this to avoid confusion?

```suggestion
    t('noteCharacterCountDescriptiveText')
      .replace('@count', characterCount.toString())
      .replace('@max', maxLength!.toString());
```

Reply:
Yes, that's nice 👌